frontend/server/workflow-helper.ts

```diff
@@ -132,19 +133,48 @@ export function toGetPodLogsStream(
 export function createPodLogsMinioRequestConfig(
   minioOptions: MinioClientOptions,
   bucket: string,
-  prefix: string,
+  keyFormat: string,
 ) {
-  // TODO: support pod log artifacts for diff namespace.
-  // different bucket/prefix for diff namespace?
-  return async (podName: string, _namespace?: string): Promise<MinioRequestConfig> => {
+  return async (podName: string, createdAt: string, namespace: string = ''): Promise<MinioRequestConfig> => {
     // create a new client each time to ensure session token has not expired
     const client = await createMinioClient(minioOptions, 's3');
-    const workflowName = workflowNameFromPodName(podName);
-    return {
-      bucket,
-      client,
-      key: path.join(prefix, workflowName, podName, 'main.log'),
-    };
+    const createdAtArray = createdAt.split("-")
+
+    let key: string = keyFormat
+      .replace(/\s+/g, '')  // Remove all whitespace.
+      .replace("{{workflow.name}}", podName.replace(/-system-container-impl-.*/, ''))
+      .replace("{{workflow.creationTimestamp.Y}}", createdAtArray[0])
+      .replace("{{workflow.creationTimestamp.m}}", createdAtArray[1])
+      .replace("{{workflow.creationTimestamp.d}}", createdAtArray[2])
+      .replace("{{pod.name}}", podName)
+      .replace("{{workflow.namespace}}", namespace)
+
+    if (!key.endsWith("/")) {
+      key = key + "/"
+    }
+    key = key + "main.log"
```

Comment:
So the `main.log` is not part of the keyFormat, but it is appended here? Maybe just adding the `main.log` to the keyFormat constant?

Reply:
Thank you so much for the feedback, @zijianjoy! 🙏

My rationale for excluding `main.log` from `keyFormat` was so that the `keyFormat` value there could precisely mirror the `keyFormat` value specified by the backend, which is set (by default) [here](https://github.com/kubeflow/pipelines/blob/7b7918ebf8c30e6ceec99283ef20dbc02fdf6a42/manifests/gcp_marketplace/chart/kubeflow-pipelines/templates/argo.yaml#L864) (internally) and [here](https://github.com/argoproj/argo-workflows/blob/main/docs/workflow-controller-configmap.yaml#L122-L127) (externally), and doesn't include `main.log`, i.e. the container name. I'm concerned that introducing a delta might confuse end users.

It looks like the `main` container name is hardcoded [here](https://github.com/kubeflow/pipelines/blob/7b7918ebf8c30e6ceec99283ef20dbc02fdf6a42/backend/src/v2/driver/driver.go#L458). Since it's hard-coded in the backend, do you still want it to be parameterized it in `config.ts`? If so, would you prefer making the user container name a dedicated parameter in `config.ts` to avoid the aforementioned `keyFormat` delta? Happy to implement either way.

Reply:
In that case, it is okay not to change the current keyFormat. We can think more about making the log name configurable, but that will be a different task. Thanks for sharing your thinking process!